Zend/zend_ini_parser.y

```diff
@@ -182,6 +182,9 @@ static void zend_ini_get_var(zval *result, zval *name)
 	} else if ((envvar = zend_getenv(Z_STRVAL_P(name), Z_STRLEN_P(name))) != NULL ||
 			   (envvar = getenv(Z_STRVAL_P(name))) != NULL) {
 		ZVAL_NEW_STR(result, zend_string_init(envvar, strlen(envvar), ZEND_SYSTEM_INI));
+	/* ..or if not defined, try fallback value */
+	} else if (fallback != NULL) {
+		ZVAL_NEW_STR(result, zend_string_init(Z_STRVAL_P(fallback), strlen(Z_STRVAL_P(fallback)), ZEND_SYSTEM_INI));
```

Comment:
This will introduce some inconsistencies where non-string literals will be treated as strings, even with `INI_SCANNER_TYPED`.

Reply:
Fixed! Will add a test that runs it typed.

Reply:
I'm now treating the fallback values similarly to offsets, so I think this is okay now. When using the typed scanner numbers correctly become numbers, falsy values become empty strings, true becomes "1", and so on.